homeassistant/components/tfl/config_flow.py

```diff
@@ -0,0 +1,284 @@
+            self.context["entry_id"]
+        )
+        assert config_entry is not None
+        if user_input is not None:
+            try:
+                await validate_app_key(self.hass, user_input[CONF_API_APP_KEY])
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                # Input is valid, set data
+                self.data[CONF_API_APP_KEY] = user_input[CONF_API_APP_KEY]
+                self.hass.config_entries.async_update_entry(
+                    config_entry, data=self.data
+                )
+                await self.hass.config_entries.async_reload(self.context["entry_id"])
+                return self.async_abort(reason="reconfigure_successful")
+
+        return self.async_show_form(
+            step_id="reconfigure",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_API_APP_KEY, default=config_entry.data[CONF_API_APP_KEY]
+                    ): cv.string,
+                }
+            ),
+            errors=errors,
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(OptionsFlow):
+    """Handles options flow for the component."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options for the TfL component."""
+        errors: dict[str, str] = {}
+        description_placeholders: dict[str, str] = {}
+
+        api_key = self.config_entry.data[CONF_API_APP_KEY]
+
+        if user_input is not None:
+            # Validate the stop points
+            errors, description_placeholders = await validate_stop_points(
+                self.hass,
+                api_key,
+                user_input[CONF_STOP_POINTS],
+            )
+            if not errors:
+                data: dict[str, Any] = {}
+                data[CONF_STOP_POINTS] = user_input[CONF_STOP_POINTS]
+
+                # Value of data will be set on the options property of our config_entry
+                # instance.
+                return self.async_create_entry(
+                    title="Transport for London",
+                    data=data,
+                )
+
+        config = self.config_entry.options
+        all_stops = deepcopy(config[CONF_STOP_POINTS])
```

Comment:
there's no need to deepcopy since we're not mutating?

Reply:
Good catch. Missed to tidy this up when I refactored to use `TextSelector`

Reply:
resolved